### PR TITLE
add huber loss function (for robust regression)

### PR DESCRIPTION
**Pull request by osh**

### CODE REVIEW

**Comment by the-moliver** [comment]
The log-cosh loss is not the same as Huber loss.

**Comment by Danielhiversen** [comment]
Yes, this is log-cosh loss.
@osh : The difference is explained here: http://www.cs.cornell.edu/courses/cs4780/2015fa/web/lecturenotes/lecturenote10.html

**Comment by osh** [comment]
thanks @the-moliver @Danielhiversen - have renamed it appropriately and fixed PEP8 checks

File: keras/losses.py [hunk 1]
```diff
@@ -33,6 +33,11 @@ def hinge(y_true, y_pred):
     return K.mean(K.maximum(1. - y_true * y_pred, 0.), axis=-1)
 
 
+def logcosh(y_true, y_pred):
+    cosh = lambda x: (K.exp(x) + K.exp(-x)) / 2
```
Comment (fchollet) [inline keras/losses.py]: Avoid naming a lambda, use a function instead

File: keras/losses.py [hunk 2]
```diff
@@ -33,6 +33,11 @@ def hinge(y_true, y_pred):
     return K.mean(K.maximum(1. - y_true * y_pred, 0.), axis=-1)
 
 
+def logcosh(y_true, y_pred):
```
Comment (fchollet) [inline keras/losses.py]: Wondering about naming. R seems to use `logcosh`. What does everyone else use?

File: keras/losses.py (hunk 1 above)
Comment (osh) [inline keras/losses.py]: should the named function remain local? or live in backend?

File: keras/losses.py (hunk 2 above)
Comment (osh) [inline keras/losses.py]: logcosh seems to be the most standard naming i can find (log_cosh seems awkward and log-cosh is right out)

**Comment by fchollet** [comment]
Le LGTM
